src/Components/Wizard/steps/Pubkeys/NewKeyForm.js

```diff
@@ -0,0 +1,113 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Form, FormGroup, TextInput, FileUpload } from '@patternfly/react-core';
+import { useWizardContext } from '../../../Common/WizardContext';
+
+// This is a simple regex format for public ssh key
+const PUBLIC_KEY_FORMAT = 'ssh-[^s]* AAAA[0-9A-Za-z+/,]+ ([^@]+@[^@]+)';
```

Comment:
ugh, this does not seem to be correct, the comment is a free text format, that does not require `@` and is optional.
https://gist.github.com/paranoiq/1932126?permalink_comment_id=3211232#gistcomment-3211232 seems to be closer, but I'd be ok keeping the first part and not get too fancy, just prefix `^(ssh|ecdsa)`

@lzap ?

Reply:
Yeah, while this is not terribly complex format (I think RFC 4716 covers it) I would definitely disencourage from doing any deep validation in the UI, at least that blocks the user in case it does not match.

If ever, I would just do "frontline" validation in here, for example only ensuring that the field contains three columns separated by whitespace. That will prevent users from accidentally submitting/pasting random text. Something like (pseudo regexp I have zero experience with JS regexp engines):

```
[^s]+\s+[^s]+\s+[^s]+
```

Reply:
I'd agree, that covering just basic format is enough, tho the last (comment) is optional I believe.

Reply:
Oh yeah then we would modify the regexp to something like `[^s]+\s+[^s]+\s*[^s]*` you get the idea.

Reply:
@amirfefer any progress here? shall we sync up and define this on slack together, or what is your plan here?

Reply:
updated :)